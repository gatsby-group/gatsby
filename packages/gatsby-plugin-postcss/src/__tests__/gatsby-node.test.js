describe(`gatsby-plugin-postcss`, () => {
  jest.mock(`../resolve`, () => module => `/resolved/path/${module}`)

  const actions = {
    setWebpackConfig: jest.fn(),
    replaceWebpackConfig: jest.fn(),
  }

  const loaders = {
    miniCssExtract: () => `miniCssExtract`,
    null: () => `null-loader`,
    css: args => `css-loader(${JSON.stringify(args)})`,
  }

  const { onCreateWebpackConfig } = require(`../gatsby-node`)

  const getConfig = jest.fn().mockReturnValue({
    module: {
      rules: [
        {
          oneOf: [
            {
              test: /\.css$/,
              loaders: [`css-loader`, `postcss-loader`],
            },
            {
              test: /\.module\.css$/,
              loaders: [`css-loader`, `postcss-loader`],
            },
          ],
        },
        {
          oneOf: [
            {
              test: /\.scss$/,
              loaders: [`css-loader`, `postcss-loader`, `sass-loader`],
            },
            {
              test: /\.module\.scss$/,
              loaders: [`css-loader`, `postcss-loader`, `sass-loader`],
            },
          ],
        },
        {
          test: /\.js/,
          use: [`babel-loader`],
        },
      ],
    },
  })

  beforeEach(() => {
    actions.setWebpackConfig.mockReset()
  })

  const tests = {
    stages: [`develop`, `build-javascript`, `develop-html`, `build-html`],
    options: {
      "No options": {},
      "PostCss options": {
        postCssPlugins: [`autoprefixer`],
        sourceMap: false,
      },
    },
  }

  tests.stages.forEach(stage => {
    for (const label in tests.options) {
      const options = tests.options[label]

      it(`Stage: ${stage} / ${label}`, () => {
        onCreateWebpackConfig({ actions, loaders, stage, getConfig }, options)

        expect(actions.setWebpackConfig).toMatchSnapshot()
      })
    }
  })
})
