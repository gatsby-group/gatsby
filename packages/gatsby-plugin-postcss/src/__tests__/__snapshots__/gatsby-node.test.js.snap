// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-postcss Stage: build-html / No options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.css\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css-loader({\\"modules\\":true,\\"importLoaders\\":1})",
                    Object {
                      "loader": "/resolved/path/postcss-loader",
                      "options": Object {
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.css\\$/,
                  "use": Array [
                    "null-loader",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
}
`;

exports[`gatsby-plugin-postcss Stage: build-html / PostCss options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.css\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css-loader({\\"modules\\":true,\\"importLoaders\\":1})",
                    Object {
                      "loader": "/resolved/path/postcss-loader",
                      "options": Object {
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.css\\$/,
                  "use": Array [
                    "null-loader",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
}
`;

exports[`gatsby-plugin-postcss Stage: build-javascript / No options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.css\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css-loader({\\"modules\\":true,\\"importLoaders\\":1})",
                    Object {
                      "loader": "/resolved/path/postcss-loader",
                      "options": Object {
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.css\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css-loader({\\"importLoaders\\":1})",
                    Object {
                      "loader": "/resolved/path/postcss-loader",
                      "options": Object {
                        "sourceMap": false,
                      },
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
}
`;

exports[`gatsby-plugin-postcss Stage: build-javascript / PostCss options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.css\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css-loader({\\"modules\\":true,\\"importLoaders\\":1})",
                    Object {
                      "loader": "/resolved/path/postcss-loader",
                      "options": Object {
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.css\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css-loader({\\"importLoaders\\":1})",
                    Object {
                      "loader": "/resolved/path/postcss-loader",
                      "options": Object {
                        "sourceMap": false,
                      },
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
}
`;

exports[`gatsby-plugin-postcss Stage: develop / No options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.css\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css-loader({\\"modules\\":true,\\"importLoaders\\":1})",
                    Object {
                      "loader": "/resolved/path/postcss-loader",
                      "options": Object {
                        "sourceMap": true,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.css\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css-loader({\\"importLoaders\\":1})",
                    Object {
                      "loader": "/resolved/path/postcss-loader",
                      "options": Object {
                        "sourceMap": true,
                      },
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
}
`;

exports[`gatsby-plugin-postcss Stage: develop / PostCss options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.css\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css-loader({\\"modules\\":true,\\"importLoaders\\":1})",
                    Object {
                      "loader": "/resolved/path/postcss-loader",
                      "options": Object {
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.css\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css-loader({\\"importLoaders\\":1})",
                    Object {
                      "loader": "/resolved/path/postcss-loader",
                      "options": Object {
                        "sourceMap": false,
                      },
                    },
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
}
`;

exports[`gatsby-plugin-postcss Stage: develop-html / No options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.css\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css-loader({\\"modules\\":true,\\"importLoaders\\":1})",
                    Object {
                      "loader": "/resolved/path/postcss-loader",
                      "options": Object {
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.css\\$/,
                  "use": Array [
                    "null-loader",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
}
`;

exports[`gatsby-plugin-postcss Stage: develop-html / PostCss options 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "module": Object {
          "rules": Array [
            Object {
              "oneOf": Array [
                Object {
                  "test": /\\\\\\.module\\\\\\.css\\$/,
                  "use": Array [
                    "miniCssExtract",
                    "css-loader({\\"modules\\":true,\\"importLoaders\\":1})",
                    Object {
                      "loader": "/resolved/path/postcss-loader",
                      "options": Object {
                        "sourceMap": false,
                      },
                    },
                  ],
                },
                Object {
                  "test": /\\\\\\.css\\$/,
                  "use": Array [
                    "null-loader",
                  ],
                },
              ],
            },
          ],
        },
      },
    ],
  ],
}
`;
